common/workunit/workunit.cpp

```diff
@@ -8618,11 +8618,20 @@ IStringIterator *CLocalWorkUnit::getLogs(const char *type, const char *instance)
 
 IPropertyTreeIterator* CLocalWorkUnit::getProcesses(const char *type, const char *instance) const
 {
-    VStringBuffer xpath("Process/%s/", type);
-    if (instance)
-        xpath.append(instance);
-    else
+    StringBuffer xpath("Process/");
+    if (isEmptyString(type) && isEmptyString(instance))
```

Comment:
apologies, I don't think this is right (my fault of suggesting it), it now conflates the purpose of the method (it can now return 'processes' or 'types'.

It was better as you had it before with a new method, but the name should have been clearer to indicate it is returning an iterator of types not processes, e.g. call it 'getProcessTypes'

Reply:
No problem. I will change back to the new method with the new name.